### PR TITLE
Feature/solver problemview

### PR DESCRIPTION
**Pull request by bj2015**

### CODE REVIEW

**Comment by coolya** [comment]
Is this a replacement for IETS3/iets3.core#344?

**Comment by itnef** [comment]
> Is this a replacement for [IETS3/iets3.core#344](https://github.com/IETS3/iets3.core/pull/344)?

No, it's a prerequisite.

**Comment by coolya** [comment]
The branch looks quite outdated. It's still using MPS 2018.3-RC2. I think we should rebase it onto current master.

File: code/languages/org.iets3.opensource/languages/org.iets3.analysis.base/models/behavior.mps [hunk 1]
```diff
@@ -2,9 +2,9 @@
 <model ref="r:729fa0c7-b4e4-42b1-acfe-71017c020a49(org.iets3.analysis.base.behavior)">
   <persistence version="9" />
   <languages>
-    <use id="af65afd8-f0dd-4942-87d9-63a55f2a9db1" name="jetbrains.mps.lang.behavior" version="2" />
-    <use id="f3061a53-9226-4cc5-a443-f952ceaf5816" name="jetbrains.mps.baseLanguage" version="8" />
-    <use id="7866978e-a0f0-4cc7-81bc-4d213d9375e1" name="jetbrains.mps.lang.smodel" version="13" />
+    <use id="af65afd8-f0dd-4942-87d9-63a55f2a9db1" name="jetbrains.mps.lang.behavior" version="-1" />
```
Comment (coolya) [inline code/languages/org.iets3.opensource/languages/org.iets3.analysis.base/models/behavior.mps]: Any idea why MPS decided to reset the language version to "unknown" in various places in the changes?

Comment (bj2015) [reply code/languages/org.iets3.opensource/languages/org.iets3.analysis.base/models/behavior.mps]: Sorry, no idea. This must have happened automatically.
Is that a problem?

**Comment by bj2015** [comment]
Don't worry. We did not intend to merge this branch without rebasing.

Rebasing is done. CI ran green.

**Comment by coolya** [comment]
> Don't worry. We did not intend to merge this branch without rebasing.

I would really appreciate if you would include such things into the description of the PR. Also including what @itnef mentioned in his comment that it's mostly interfaces changes and no implementation helps to review this. The more valuable context you provide for a  PR the better others can review it and speeds up things.

File: code/languages/org.iets3.opensource/languages/org.iets3.analysis.base/models/behavior.mps (hunk 1 above)
Comment (coolya) [reply code/languages/org.iets3.opensource/languages/org.iets3.analysis.base/models/behavior.mps]: Most  probably as soon as anyone changes these languages MPS will update the language versions which will then trigger a "update module descriptors" migration. It's not a really problem I think.

**Comment by bj2015** [comment]
@coolya Regarding the inference rule: You are absolutely right. When it does not contribute any constraints, it should of course be a checking rule. Fixed.

**Comment by coolya** [comment]
Apparently these changes broke the core build. 

https://build.mbeddr.com/viewLog.html?buildId=235782&buildTypeId=Iets3_Core&tab=buildLog&branch_Iets3_Iets3Core=%3Cdefault%3E

**Pull request merged**